### ISSUE TITLE
 Loss_bbox is negative using pytorch-style backbone for FCOS

### ISSUE THREAD

**Issue by AAA-Fan** [open-mmlab/mmdetection#3900]
My config:
```
_base_ = [
    '../_base_/datasets/coco_detection.py',
    '../_base_/schedules/schedule_1x.py', '../_base_/default_runtime.py'
]
# model settings
model = dict(
    type='FCOS',
    #pretrained='open-mmlab://detectron/resnet50_caffe',
    pretrained="/mnt/nas4/users/tangyujia/code/mmdetection-v2/checkpoints/resnet50-19c8e357.pth",
    backbone=dict(
        type='ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        norm_cfg=dict(type='BN', requires_grad=False),
        norm_eval=True,
        style='pytorch'),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        start_level=1,
        add_extra_convs=True,
        extra_convs_on_inputs=False,  # use P5
        num_outs=5,
        relu_before_extra_convs=True),
    bbox_head=dict(
        type='FCOSHead',
        num_classes=80,
        in_channels=256,
        stacked_convs=4,
        feat_channels=256,
        strides=[8, 16, 32, 64, 128],
        norm_cfg=None,
        loss_cls=dict(
            type='FocalLoss',
            use_sigmoid=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
        loss_bbox=dict(type='IoULoss', loss_weight=1.0),
        loss_centerness=dict(
            type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0)))
# training and testing settings
train_cfg = dict(
    assigner=dict(
        type='MaxIoUAssigner',
        pos_iou_thr=0.5,
        neg_iou_thr=0.4,
        min_pos_iou=0,
        ignore_iof_thr=-1),
    allowed_border=-1,
    pos_weight=-1,
    debug=False)
test_cfg = dict(
    nms_pre=1000,
    min_bbox_size=0,
    score_thr=0.05,
    nms=dict(type='nms', iou_threshold=0.5),
    max_per_img=100)
img_norm_cfg = dict(
    mean=[102.9801, 115.9465, 122.7717], std=[1.0, 1.0, 1.0], to_rgb=False)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='Resize', img_scale=(1333, 800), keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels']),
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(1333, 800),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(type='Normalize', **img_norm_cfg),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img']),
        ])
]
data = dict(
    samples_per_gpu=4,
    workers_per_gpu=4,
    train=dict(pipeline=train_pipeline),
    val=dict(pipeline=test_pipeline),
    test=dict(pipeline=test_pipeline))
# optimizer
optimizer = dict(
    lr=0.01, paramwise_cfg=dict(bias_lr_mult=2., bias_decay_mult=0.))
optimizer_config = dict(
    _delete_=True, grad_clip=dict(max_norm=35, norm_type=2))
# learning policy
lr_config = dict(
    policy='step',
    warmup='constant',
    warmup_iters=500,
    warmup_ratio=1.0 / 3,
    step=[8, 11])
total_epochs = 12
```

The log I got:

2020-10-09 16:11:14,467 - mmdet - INFO - load model from: /mnt/nas4/users/tangyujia/code/mmdetection-v2/checkpoints/resnet50-19c8e357.pth
2020-10-09 16:11:14,801 - mmdet - WARNING - The model and loaded state dict do not match exactly

unexpected key in source state_dict: fc.weight, fc.bias

loading annotations into memory...
Done (t=35.28s)
creating index...
index created!
fatal: Not a git repository (or any parent up to mount point /mnt/nas4)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
loading annotations into memory...
Done (t=1.19s)
creating index...
index created!
2020-10-09 16:12:01,359 - mmdet - INFO - Start running, host: tangyujia@aarc55009, work_dir: /mnt/nas4/users/tangyujia/code/mmdetection-v2/work_dirs/fcos_                                        r50_caffe_fpn_4x4_1x_coco
2020-10-09 16:12:01,359 - mmdet - INFO - workflow: [('train', 1)], max: 12 epochs
2020-10-09 16:12:27,254 - mmdet - INFO - Epoch [1][50/29317]    lr: 3.333e-03, eta: 2 days, 2:34:46, time: 0.518, data_time: 0.072, memory: 5859, loss_cls                                        : 1.2029, loss_bbox: -13.9699, loss_centerness: 0.6899, loss: -12.0771, grad_norm: nan
2020-10-09 16:12:50,030 - mmdet - INFO - Epoch [1][100/29317]   lr: 3.333e-03, eta: 1 day, 23:32:14, time: 0.456, data_time: 0.015, memory: 5859, loss_cls                                        : 1.0893, loss_bbox: -23.9148, loss_centerness: 0.6661, loss: -22.1594, grad_norm: nan
2020-10-09 16:13:12,082 - mmdet - INFO - Epoch [1][150/29317]   lr: 3.333e-03, eta: 1 day, 22:02:51, time: 0.441, data_time: 0.013, memory: 5859, loss_cls                                        : 1.0314, loss_bbox: -24.0113, loss_centerness: 0.6648, loss: -22.3151, grad_norm: nan
2020-10-09 16:13:34,886 - mmdet - INFO - Epoch [1][200/29317]   lr: 3.333e-03, eta: 1 day, 21:40:00, time: 0.456, data_time: 0.015, memory: 5859, loss_cls                                        : 1.0042, loss_bbox: -24.0223, loss_centerness: 0.6657, loss: -22.3524, grad_norm: nan
2020-10-09 16:13:57,034 - mmdet - INFO - Epoch [1][250/29317]   lr: 3.333e-03, eta: 1 day, 21:10:46, time: 0.443, data_time: 0.015, memory: 5859, loss_cls                                        : 0.9692, loss_bbox: -24.0146, loss_centerness: 0.6641, loss: -22.3814, grad_norm: nan
2020-10-09 16:14:19,522 - mmdet - INFO - Epoch [1][300/29317]   lr: 3.333e-03, eta: 1 day, 20:57:47, time: 0.450, data_time: 0.016, memory: 5859, loss_cls                                        : 0.9830, loss_bbox: -24.0631, loss_centerness: 0.6646, loss: -22.4155, grad_norm: nan
2020-10-09 16:14:41,932 - mmdet - INFO - Epoch [1][350/29317]   lr: 3.333e-03, eta: 1 day, 20:47:07, time: 0.448, data_time: 0.015, memory: 5859, loss_cls                                        : 0.9178, loss_bbox: -23.7964, loss_centerness: 0.6630, loss: -22.2156, grad_norm: nan

**Comment by yuzhj** [comment]
a smaller learning rate, such as 0.005

**Comment by ZwwWayne** [comment]
You need to modify img_norm_cfg and set `norm_cfg=dict(type='BN', requires_grad=True)` for pytorch style backbone.
